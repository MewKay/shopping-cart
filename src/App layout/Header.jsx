import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const Header = ({ totalQuantity = 0, handleShowCartModal }) => {
  return (
    <header>
      <h1>My Shop App</h1>
      <nav>
        <Link to={"/"}>Home</Link>
        <Link to={"store"}>Store</Link>
        <button onClick={handleShowCartModal}>Cart({totalQuantity})</button>
      </nav>
    </header>
  );
};

Header.propTypes = {
  totalQuantity: PropTypes.number,
  handleShowCartModal: PropTypes.func.isRequired,
};

export default Header;
