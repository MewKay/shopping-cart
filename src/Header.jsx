import PropTypes from "prop-types";
import { Link } from "react-router-dom";

const Header = function createHeaderComponent({ itemNumber = 0 }) {
  return (
    <header>
      <h1>My Shop App</h1>
      <nav>
        <Link to={"/"}>Home</Link>
        <Link to={"store"}>Store</Link>
        <a>Cart({itemNumber})</a>
      </nav>
    </header>
  );
};

Header.propTypes = {
  itemNumber: PropTypes.number,
};

export default Header;
