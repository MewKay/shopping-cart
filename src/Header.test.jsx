import { describe, expect, it } from "vitest";
import { render } from "@testing-library/react";
import Header from "./Header";
import { BrowserRouter } from "react-router-dom";

describe("Header component", () => {
  it("renders the brand name and the nav links", () => {
    const { container } = render(<Header />, { wrapper: BrowserRouter });

    expect(container).toMatchSnapshot();
  });
});
