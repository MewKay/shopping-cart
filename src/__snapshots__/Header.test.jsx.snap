// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header component > renders the brand name and the nav links 1`] = `
<div>
  <header>
    <h1>
      My Shop App
    </h1>
    <nav>
      <a
        href="/"
      >
        Home
      </a>
      <a
        href="/store"
      >
        Store
      </a>
      <a>
        Cart(
        0
        )
      </a>
    </nav>
  </header>
</div>
`;
